src/templates/React InstantSearch widget/src/lib/component.tsx.hbs

```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+type Props = {
+  // TODO: this is the object returned by `getProvidedProps` from the connector.
+};
+
+export const {{ pascalCaseName }}Component = ({}: Props) => {
```

Comment:
What do we want to do? Here the `Props` is loose. It's not connected with the return of `getProvidedProps` from the connector. Do we want to type this template at that level? Or this is okay enough?

Reply:
can it not be imported from the connector file?

Reply:
The return type of getProvidedProps won't include `refine` function, but the `Props` here in the component will. And we might want to specify in the connector as a generic, what kind of type the `refine` function will take. So these thoughts kept leading to another, and I wonder maybe the current version is incomplete but simple to start with.

Reply:
could it be ReturnType<getProvidedProps> & { refine } or something?

Reply:
![CleanShot 2021-06-08 at 16 44 05@2x](https://user-images.githubusercontent.com/499898/121206610-e6dff600-c878-11eb-9bab-33f2fa430759.png)

what do you think?

Reply:
https://github.com/algolia/create-instantsearch-app/pull/456/commits/dec9c2ad210b65ec1a11b714f74e78e1d343f755